src/lib.rs

```diff
@@ -72,21 +72,19 @@ where
 
     /// Read enough registers to fill our read buf
     pub fn read_registers(&mut self) -> Result<(), Error<CommE, PinE>> {
+        let read_reg = [Self::REG_FIRST; 1];
         self.i2c
-            .write_read(
-                Self::DEFAULT_I2C_ADDRESS,
-                &[Self::REG_FIRST],
-                self.blob_buf.as_mut(),
-            )
+            .write_read(Self::DEFAULT_I2C_ADDRESS, &read_reg, self.blob_buf.as_mut())
             .map_err(Error::Comm)?;
         Ok(())
     }
 
     pub fn read_truncated_registers(&mut self) -> Result<(), Error<CommE, PinE>> {
+        let read_reg = [Self::REG_FIRST; 1];
```

Comment:
This was the actual change necessary for the latest `nrf52832-hal` to work. It [checks if memory is from RAM](https://github.com/nrf-rs/nrf-hal/blob/master/nrf-hal-common/src/twim.rs#L239-L241) for some reason.